core-jdk8/src/main/java/org/mapstruct/ValueMapping.java

```diff
@@ -0,0 +1,125 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Configures the mapping of source constant value to target constant value.
+ * <p>
+ * Supported mappings are
+ * <ol>
+ * <li>Enumeration to Enumeration</li>
+ * </ol>
+ * <p>
+ * <B>Example 1:</B>
+ * <pre>
+ * <code>
+ * public enum OrderType { RETAIL, B2B, EXTRA, STANDARD, NORMAL }
+ *
+ * public enum ExternalOrderType { RETAIL, B2B, SPECIAL, DEFAULT }
+ *
+ * &#64;ValueMappings({
+ *    &#64;ValueMapping(source = "EXTRA", target = "SPECIAL"),
+ *    &#64;ValueMapping(source = "STANDARD", target = "DEFAULT"),
+ *    &#64;ValueMapping(source = "NORMAL", target = "DEFAULT")
+ * })
+ * ExternalOrderType orderTypeToExternalOrderType(OrderType orderType);
+ * </code>
+ * Mapping result:
+ * ╔═════════════════════╦════════════════════════════╗
+ * ║ OrderType           ║ ExternalOrderType          ║
+ * ╠═════════════════════╬════════════════════════════╣
+ * ║ &lt;NULL&gt;              ║ &lt;NULL&gt;                     ║
```

Comment:
Yes, I just mean the wording in the javadoc here. You want to describe what input creates what output in the generated mapping method, and `"<NULL>"` is not the input here, but `null` is. Or did I misunderstand the meaning of that table?